### DISCUSSION TITLE
Allow users to change the date source for the calendar

### DISCUSSION THREAD

**Discussion by stevejburge** [publishpress/PublishPress-Planner#1340] [Feature Requests]
One request to use dates from ACF:

> We'd like to use the calendar to manage upcoming Webinar events. Unfortunately the calendar only shows the post by publish date not the event date. Is there a way to have the calendar list posts based on a custom date field rather than the publish date? I'm using ACF plugin (Advanced Custom Fields) built in date picker.
> 
> I'm using another calendar on the site called Calendar Anything that allows me to hook into custom post types (As yours does) but also custom fields. 

This is the referenced plugin: https://www.inkthemes.com/market/calendar-wordpress-plugin/

Perhaps related is this request to show some posts on the calendar based on a PublishPress metadata field:

> Is there any other way to get the blogs in Assigned status to show up with the date the first draft is due (set in metadata or Publish date)? It's an important part of our workflow and it seems it should be accessible through the calendar in addition to publishing dates.

We also have a request to use the modified date: https://github.com/publishpress/PublishPress/issues/561

**Comment by andergmartins** [comment]
@stevejburge, here is a proposal for the field to change the date source:

<img width="1283" alt="Screen Shot 2021-04-16 at 17 52 49" src="https://user-images.githubusercontent.com/407131/115082731-ba44d900-9edc-11eb-8a72-acac19e624ea.png">

We could display on the select field:

* Publishing date
* Modifying date
* Date metada fields (individually)

And maybe have a setting field to specify what post metadata fields could be used as date source

**Comment by stevejburge** [comment]
Thanks @andergmartins

- Would this be difficult?
- Should it be in "Settings"?

**Comment by andergmartins** [comment]
Considering the calendar is now asynchronous I think we could have a setting for selecting the default field and maybe other options, but I think we could keep the select box on the calendar page allowing to alternate between the date used to organize the calendar.

About the difficulty, it is basically changing the query to filter by another field, but I expect having some extra work to optimize the query, specially if filtering by a metadata field. For creating the settings and new query, I think the required timeframe is something like 2 days.

**Comment by stevejburge** [comment]
Thanks @andergmartins. Let's file under "nice to have in the future". No need to add it to a milestone yet
